modules/jenkins.js

```diff
@@ -0,0 +1,90 @@
+// -----------------------------------------------------------------------------------------
+// #region Imports
+// -----------------------------------------------------------------------------------------
+
+const {
+    StringUtils,
+    CollectionUtils,
+} = require("andculturecode-javascript-core");
+const echo = require("./echo");
+const fs = require("fs");
+const os = require("os");
+const path = require("path");
+const upath = require("upath");
+
+// #endregion Imports
+
+// -----------------------------------------------------------------------------------------
+// #region Constants
+// -----------------------------------------------------------------------------------------
+const CONFIG_FILE = ".jenkinsconfig";
+const BASE_CONFIG = {
+    url: "",
+    username: "",
+    token: "",
+    profiles: {},
+};
+
+// #endregion Constants
+
+// -----------------------------------------------------------------------------------------
+// #region Functions
+// -----------------------------------------------------------------------------------------
+
+const jenkins = {
+    configureCredentials(url, username, token) {
+        const errors = [];
+        if (StringUtils.isEmpty(url)) {
+            errors.push("Url is required");
+        }
+        if (StringUtils.isEmpty(username)) {
+            errors.push("Username is required");
+        }
+        if (StringUtils.isEmpty(token)) {
+            errors.push("Api token is required");
+        }
+        if (CollectionUtils.hasValues(errors)) {
+            echo.errors(errors);
+            return false;
+        }
+        let config = BASE_CONFIG;
+        Object.assign(config, { url, username, token });
+        this.writeToConfig(config);
```

Comment:
Should this function forward the return value of `this.writeToConfig` call? Might be misleading that the credentials file was configured correctly if the file couldn't save for some reason.